core/modules/rproc/InfileMerger.cc

```diff
@@ -261,19 +241,18 @@ bool InfileMerger::merge(std::shared_ptr<proto::WorkerResponse> response) {
     if (response->result.row_size() == 0) {
         return true;
     }
-    _sizeCheckRowCount += response->result.row_size();
-
 
-    SemaLock semaLock(semaMgrLocal); // &&&
+    std::unique_ptr<util::SemaLock> semaLock;
+    if (_dbEngine != MYISAM) {
```

Comment:
Based on subsequent comment that "// needed for INNODB and MYISAM", did you mean to compare with `MEMORY` instead?```
    if (_dbEngine == MEMORY) {
        // needed for INNODB and MYISAM
        ...
```

Reply:
Typo, should be   // needed for INNODB and MEMORY. 

MyISAM opens one connection when the user query starts and reuses for every merge. INNODB and MEMORY have parallel merges and make new connections as needed and need SemaLock to keep from opening too many connections to MySQL.